llmstudio/engine/providers/azure.py

```diff
@@ -44,48 +56,284 @@ async def generate_client(
+                                    arguments="",
+                                    type="function",
+                                ),
+                            )
+                        ],
+                    ),
+                    finish_reason=None,
+                    index=0,
+                )
+            ],
+            created=int(time.time()),
+            model=kwargs.get("request").model,
+            object="chat.completion.chunk",
+        ).model_dump()
+
+    def create_tool_finish_chunk(self, kwargs):
+        return ChatCompletionChunk(
+            id=str(uuid.uuid4()),
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(),
+                    finish_reason="tool_calls",
+                    index=0,
+                )
+            ],
+            created=int(time.time()),
+            model=kwargs.get("request").model,
+            object="chat.completion.chunk",
+        ).model_dump()
+
+    def create_tool_argument_chunk(self, content, kwargs):
+        return ChatCompletionChunk(
+            id=str(uuid.uuid4()),
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(
+                        tool_calls=[
+                            ChoiceDeltaToolCall(
+                                index=0,
+                                function=ChoiceDeltaToolCallFunction(
+                                    arguments=content,
+                                ),
+                            )
+                        ],
+                    ),
+                    finish_reason=None,
+                    index=0,
+                )
+            ],
+            created=int(time.time()),
+            model=kwargs.get("request").model,
+            object="chat.completion.chunk",
+        ).model_dump()
+
+    def create_tool_first_chunk(self, kwargs):
+        return ChatCompletionChunk(
+            id=str(uuid.uuid4()),
+            choices=[
+                Choice(
+                    delta=ChoiceDelta(
+                        content=None,
+                        function_call=None,
+                        role="assistant",
+                        tool_calls=None,
+                    ),
+                    index=0,
+                )
+            ],
+            created=int(time.time()),
+            model=kwargs.get("request").model,
+            object="chat.completion.chunk",
+            usage=None,
+        ).model_dump()
+
+    def convert_to_openai_format(self, message):
+        if isinstance(message, str):
+            return [{"role": "user", "content": message}]
+        return message
+
+    def add_system_message(self, openai_message, llama_message, tools):
+        system_message = ""
+        system_message_found = False
+        for message in openai_message:
+            if message["role"] == "system" and message["content"] is not None:
+                system_message_found = True
```

Comment:
this flag will never be False again. is it compromising the condition on line 260?